### PR TITLE
PR: Update coveralls config (CI)

### LINKED ISSUE #426: Coveralls config needs to be updated
Not totally sure why but seems like the coverage reports started to be handled differently. After doing a merge with master a bunch of failed checks appeared: https://github.com/spyder-ide/qtpy/commit/bc7a3c746c88ea3e5d8f02818763636bae807523 

Checking the coveralls page seems like now the coverage is associated to the path where the report is located instead of being aggregated?:

image

Could this be related with the change to use mamba or could it be related with some updates on the Coveralls tools that require for us to update the coverage upload config? What do you think @CAM-Gerlach?

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #426

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -125,9 +125,19 @@ jobs:
...
           cd temp_test_dir  # Switch to test working dir per non-src-layout hack
           cat qtpy_basedir.txt
-          pipx run coveralls --service=github --rcfile="../.coveragerc" --basedir="$(cat qtpy_basedir.txt)"
+          pipx run coveralls --service=github --rcfile="../.coveragerc" --basedir="../qtpy"
+  finish:
+    needs: test
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Coveralls Finished
```
Comment (CAM-Gerlach) [inline .github/workflows/ci.yml:140]:
```suggestion
    name: Finish Coveralls
    needs: test
    if: ${{ always() }}
    runs-on: ubuntu-latest
    steps:
    - name: Mark Coveralls finished
```

Teak human-readable names to be more appropriate and descriptive

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,9 +125,19 @@ jobs:
...
+      uses: coverallsapp/github-action@v2
+      with:
+        parallel-finished: true
```
Comment (CAM-Gerlach) [inline .github/workflows/ci.yml:143]: Missing EOF EOL here

**Review by CAM-Gerlach — COMMENTED** [review]
A couple minor comments

**Review by ccordoba12 — APPROVED** [review]
Looks good to me, thanks @dalthviz!

**Comment by dalthviz** [comment]
Is okay if we merge this one @CAM-Gerlach ? Or do you think something is missing?

**Comment by dalthviz** [comment]
Going ahead and merging this. However, if there are things that needed to be further worked on, let me know

**Pull request merged**